# Análise dos dados de viagem dos táxis amarelos de New York

In [1]:
# import das bibliotecas e funções necessárias
import pandas as pd

## Em que período do ano os táxis são mais utilizados?

In [2]:
# Import para conversão dos meses de número para string
import calendar

def most_trips():
    # Lista para armazenar as viagens de taxis por mês
    trip_month = []

    # Especificação dos tipos das colunas para diminuir o consumo de memória da função read_csv()
    column_type = {'VendorID': float, 'tpep_pickup_datetime': str, 'tpep_dropoff_datetime':  str, \
                'passenger_count': float, 'trip_distance': float, 'RatecodeID': float, \
                'store_and_fwd_flag': str, 'PULocationID': int, 'DOLocationID': int, \
                'payment_type': float, 'fare_amount': float, 'extra': float, 'mta_tax': float, \
                'tip_amount': float, 'tolls_amount': float, 'improvement_surcharge': float, \
                'total_amount': float, 'congestion_surcharge': float}
    
    for i in range(1, 13):
        # Carregamento do dataset referente ao mês da iteração
        month = pd.read_csv('data/yellow_tripdata_2020-' + str(i).zfill(2) + '.csv', dtype = column_type)
        
        # Adiciona o número de viagens na lista trip_month
        trip_month.append([int(str(i).zfill(2)), month['tpep_pickup_datetime'].count()])
        
        # Remove o dataset do mês a fim de liberar espaço na memória
        del month
    
    # Cria um dataset para armazenar as viagens por mês a partir da lista trip_month
    number_trips = pd.DataFrame(trip_month, columns = ['month', 'number_trips'])
    
    # Converte o número referente ao mês em nome do mês para fins estéticos
    number_trips['month'] = number_trips['month'].apply(lambda x: calendar.month_name[x])
    
    return number_trips


most_trips().head(12)

month  number_trips
0     January       6405008
1    February       6299354
2       March       3007292
3       April        237993
4         May        348371
5        June        549760
6        July        800412
7      August       1007284
8   September       1341012
9     October       1681131
10   November       1508985
11   December       1461897

### Os taxis foram mais utilizados no mês de Janeiro de 2020, com um número de 6405008 viagens.

## Demonstre através de gráficos os bairros que possuem maior uso de táxis por mês.

## Em quais janela de tempo as viagens contém mais passageiros?

## Demonstre através de gráficos os bairros que possuem maior número de passageiros por janela de tempo.
Utilize as janelas abaixo:

 - 0 as 5 (4:59)
 - 5 as 11
 - 11 as 14
 - 14 as 17
 - 17 as 21
 - 21 as 24

## Como a distância se correlaciona com a duração das viagens? 
Demonstre através de gráficos essa correlação.

## Descreva como você faria para agendar a execução diaria e/ou mensal destes jobs.